cli/tests/test_mcp_dns_pinning.py

```
@@ -0,0 +1,178 @@
+            if not thread.is_alive():
+                raise RuntimeError("MCP test server exited during startup")
+            if time.monotonic() >= deadline:
+                raise TimeoutError("MCP test server did not start")
+            time.sleep(0.01)
+        yield port
+    finally:
+        server.should_exit = True
+        thread.join(timeout=5)
+        listener.close()
+        if thread.is_alive():
+            raise RuntimeError("MCP test server did not stop")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)